cassandra/metadata.py

```diff
@@ -450,37 +450,82 @@ def make_token_replica_map(self, token_to_host_owner, ring):
         return {}
 
 
-def parse_replication_factor(input_rf):
+class ReplicationFactor(object):
     """
-    Given the inputted replication factor, returns a tuple containing number of total replicas
-    and number of transient replicas
+    Represent the replication factor of a keyspace.
     """
+
+    all_replicas = None
+    """
+    The number of total replicas.
+    """
+
+    full_replicas = None
+    """
+    The number of replicas that own a full copy of the data. This is the same
+    than `all_replicas` when transient replication is not enabled.
+    """
+
     transient_replicas = None
-    try:
-        total_replicas = int(input_rf)
-    except ValueError:
+    """
+    The number of transient replicas.
+
+    Only set if the keyspace has transient replication enabled.
+    """
+
+    def __init__(self, all_replicas, transient_replicas=None):
+        self.all_replicas = all_replicas
+        self.transient_replicas = transient_replicas
+        self.full_replicas = (all_replicas - transient_replicas) if transient_replicas else all_replicas
+
+    @staticmethod
+    def create(rf):
+        """
+        Given the inputted replication factor string, parse and return the ReplicationFactor instance.
+        """
+        transient_replicas = None
         try:
-            rf = input_rf.split('/')
-            total_replicas, transient_replicas = int(rf[0]), int(rf[1])
-        except Exception:
-            raise ValueError("Unable to determine replication factor from: {}".format(input_rf))
-    return total_replicas, transient_replicas
+            all_replicas = int(rf)
+        except ValueError:
+            try:
+                rf = rf.split('/')
+                all_replicas, transient_replicas = int(rf[0]), int(rf[1])
+            except Exception:
+                raise ValueError("Unable to determine replication factor from: {}".format(rf))
+
+        return ReplicationFactor(all_replicas, transient_replicas)
+
+    def __str__(self):
+        return ("%d/%d" % (self.all_replicas, self.transient_replicas) if self.transient_replicas
+                else "%d" % self.all_replicas)
+
+    def __eq__(self, other):
+        if not isinstance(other, ReplicationFactor):
+            return False
+
+        return self.all_replicas == other.all_replicas and self.full_replicas == other.full_replicas
 
 
 class SimpleStrategy(ReplicationStrategy):
 
-    replication_factor = None
-    """
-    The replication factor for this keyspace.
+    replication_factor_info = None
```

Comment:
`info` anything better to suggest?

Reply:
`info` is ok. I also thought maybe `full_replication_factor` or `complete_replication_factor`.

Do we need to expose this publicly? It'll be shown in the `export_as_string` methods. Since there's not really a consistent interface between the different ReplicationFactor implementations, I think it makes more sense to keep it private. I'd lean towards removing the class variable and just have a private `_replication_factor` instance variable.

Reply:
The problem I see if we don't expose it is that for a transient replication, we are exposing an *incomplete* RF (since replication_factor only expose full_replicas). A user cannot even know if the rf is transient or not. export_as_string is not really convenient except for generating cql.